observability/opencensus/v2/client/client_test.go

```diff
@@ -215,6 +343,49 @@ type fakeHandler struct {
 	requests []requestValidation
 }
 
+type testExporter struct {
+	spans []*trace.SpanData
+}
+
+func (t *testExporter) ExportSpan(s *trace.SpanData) {
+	t.spans = append(t.spans, s)
+}
+
+type fakeObservabilityServiceWithError struct{}
+
+func (n fakeObservabilityServiceWithError) InboundContextDecorators() []func(context.Context, binding.Message) context.Context {
+	return nil
+}
+
+func (n fakeObservabilityServiceWithError) RecordReceivedMalformedEvent(ctx context.Context, err error) {
+}
+
+func (n fakeObservabilityServiceWithError) RecordCallingInvoker(ctx context.Context, event *event.Event) (context.Context, func(errOrResult error)) {
```

Comment:
why not just name it `err` instead of errOrResult?

Reply:
```suggestion
func (n fakeObservabilityServiceWithError) RecordCallingInvoker(ctx context.Context, event *event.Event) (context.Context, func(error)) {
```
Also, the return parameter definition does not need a variable name here.

Reply:
Forget my first comment, the way the error is wrapped in the cloudevents sdk seems rather un-go-like to me. I think you could use the `Result` class (https://github.com/cloudevents/sdk-go/blob/main/v2/protocol/result.go#L14) instead of `error`, which might make more sense to people looking at this.

Reply:
Yeah it is a bit unclear the types. The errOrResult here means that it could be a go "standard" error, or a [transport.Result type from cloudevents](https://github.com/cloudevents/sdk-go/blob/70febdc855835091a1d5f111d9223f226eb0ab0a/v2/client/invoker.go#L86) I *guess* it could use the `transport.Result` but I'm not confident in changing it 😅

It is also part of the ObservabilityService interface and what they also use in their opencensus impl: https://github.com/cloudevents/sdk-go/blob/main/observability/opencensus/v2/client/observability_service.go @dyladan what do you think?

Reply:
I'll resolve this as I think changing the API is out of scope of this.